yamllint/rules/quoted_strings.py

```diff
@@ -45,13 +72,23 @@
 
 ID = 'quoted-strings'
 TYPE = 'token'
-CONF = {'quote-type': ('any', 'single', 'double')}
-DEFAULT = {'quote-type': 'any'}
+CONF = {'quote-type': ('any', 'single', 'double'),
+        'required': (True, False, 'only-when-needed')}
+DEFAULT = {'quote-type': 'any',
+           'required': True}
 
+DEFAULT_SCALAR_TAG = u'tag:yaml.org,2002:str'
+START_TOKENS = {'#', '*', '!', '?', '@', '`', '&',
```

Comment:
I wondered why you included `@` and `, but you're right:
> The “@” (#x40, at) and “`” (#x60, grave accent) are reserved for future use.

-- https://yaml.org/spec/1.2/spec.html#id2774058